components/sup/src/supervisor.rs

```diff
@@ -313,8 +309,8 @@ impl Supervisor {
 
         unsafe {
```

Comment:
You probably should be able to remove this `unsafe` block in favor of the more scoped block in core.